### DISCUSSION TITLE
Links to accept or withdraw cookie consent requests remove current query string

### DISCUSSION THREAD

**Discussion by robsonsobral** [ExpressionEngine/ExpressionEngine#3380] [Feature Requests]
I guess the title says everything.

**Comment by intoeetive** [comment]
I'm not sure the title tells me everything =)

Did you intend to say that after clicking the link generated with `{consent_grant_url return='preferences/saved'}` the user is redirected to the homepage instead of `preferences/saved`?

**Comment by robsonsobral** [comment]
Oops! Sorry! 🤭

Nopes! I'm saying that, in case the current URI is `/foo/bar?something-to-use-on-request-module=item`, the user is redirected to `/foo/bar`.

**Comment by intoeetive** [comment]
Did you specify anything in `return` parameter? Also, does this happen only when using the link, or also with the form? (if you have checked this, if not - no worries)

**Comment by robsonsobral** [comment]
No `return` parameter! I'm sorry, but I checked the links only.

**Comment by intoeetive** [comment]
Quick investigation shows there's no easy & reliable way we could make that work though different environments. 

So one would need to use `return` parameter for such cases.

Will keep the issue open however for time-consuming but reliable implementation

**Comment by intoeetive** [comment]
Since this would rather be an enhancement, converting to discussion
